core/src/test/scala/org/apache/spark/internal/config/ConfigEntrySuite.scala

```diff
@@ -128,6 +128,25 @@ class ConfigEntrySuite extends SparkFunSuite {
     assert(conf.get(transformationConf) === "bar")
   }
 
+  test("conf entry: checkValue()") {
+    def createConf(default: Int): ConfigEntry[Int] =
+      ConfigBuilder(testKey("checkValue"))
+        .intConf
+        .checkValue(value => value >= 0, "value must be non-negative")
+        .createWithDefault(default)
+
+    // this succeeds
+    val conf = createConf(10)
+
+    // this fails because valueConverter() calls checkValue()
+    val e1 = intercept[IllegalArgumentException] { conf.valueConverter("-1") }
```

Comment:
thanks, i was about to explain this after tests pass

there exists this problem:
here in the `core` module, conf.set(...) does not call `valueConverter` to do sanity checks(see https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkConf.scala#L98)
so `conf.set(...)` won't fail while we expect it to fail.

however in `sql/core` module, `conf.set(...)` does call `entry.valueConverter(value)`(see https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L893)
so I added a test in `SQLConfEntrySuite.scala` below, to verify that it does fail for `conf.set(...)`

should we fix it (do sanity-checks for `conf.set(...)` in module `spark/core`) in this PR ? your thoughts please?